### PR TITLE
Remove deprecated settings implementation

### PR DESCRIPTION
**Pull request by youngmit**
This removes the old settings module and renames setting2.py to
setting.py. This is a breaking change for plugins that were doing
something like `from armi.settings import setting2 as setting`, but
trivial to refactor around.

### CODE REVIEW

File: armi/settings/setting.py [hunk 1]
```diff
@@ -98,431 +175,132 @@ def value(self):
...
         """
-        attrib = Setting.getDefaultAttributes(self)
-        attrib["containedType"] = self.containedType
-        attrib["options"] = self.options
-        attrib["enforcedOptions"] = self.enforcedOptions
-        return attrib
-
-
-# help direct python types to the respective setting type
-SUBSETTING_MAP = {
```
Comment (ntouran) [inline armi/settings/setting.py:520]: ugh this diff renders weirdly rather than showing a rename.

**Comment by youngmit** [comment]
Yeah, should have looked at that coverage stuff earlier. I'll comb through the coveralls stuff and see what I can do

File: armi/settings/tests/test_settingsIO.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 TerraPower, LLC
```
Comment (ntouran) [inline armi/settings/tests/test_settingsIO.py:1]: :open_mouth:

Comment (youngmit) [reply armi/settings/tests/test_settingsIO.py:1]: Eyes like hawk

**Comment by youngmit** [comment]
I think 0.08% is pretty good

**Pull request merged**